### ISSUE TITLE
Some code questions 

### ISSUE THREAD

**Issue by axxx-xxxa** [huixiancheng/SENet#22]
Thanks for the sharing code.
1. I only saw the repvgg module in the code, but you mentioned it in your paper without actually using it, right?
2. It seems like the inference model only outputs [out] instead of [out, res_2, res_3, res_4] in the code. It looks like I could potentially trim off the extra three output heads when converting it to trt.
3. Are the first three convolutional layers used for upsampling XYZIR, similar to the upsampling MLP for XYZ in PointNet?

btw
The CENet with 64x512 input model has been deployed on Jetson AGX Xavier and it worked successfully.
fp32 ≈ 6HZ
fp16 ≈10HZ

**Comment by huixiancheng** [comment]
Hi. Thx for your attention. For ques:

1.  Yes. I did not use any repvgg modules for this job. I'm just quoting their experimental description to show the efficiency comparison between 1x1 conv and 3x3 conv. However, I think it might be interesting to explore re-parameterization architectures for LiDAR segmentation in a range view perspective.
2.  Yes, you are right, this part should be discarded when inferring.
3.  The first three convolutional layers are only used to extract features and increase dimensions.

Glad to hear that 👍, network design on constrained devices is also a good topic.
In my personal opinion, changing [the first downsampling](https://github.com/huixiancheng/CENet/tree/main/modules/network/ResNet.py#L111-L114) to 2x downsampling gives better inference speed (i.e. 16x downsampling overall instead of the original 8x), however such a choice would require retraining or fine-tuning the model, and the final performance would be reduced.
Moreover, MINet ([Code](https://github.com/sj-li/MINet) and [Paper](https://arxiv.org/abs/2008.09162)) also provides speed tests on the Jetson AGX, which might be a good reference.

**Comment by axxx-xxxa** [comment]
还想向您请教几个问题：
1. 我发现基于depth排序后将点放入特征图中，会导致同一角度远处点受近处点影响(同角度多点只取最近点分类结果)，这是基于spherical投影的局限性吗？增大特征图可以应该可以解决部分这种问题吧，如果增大特征图的话，H,W同时增大有效吗？因为大多数网络都是只修改W的大小。
2. 上次说到32线/64线鲁棒性的问题，我打算将kitti数据集抽成32线进行测试，不知道这样是不是有效的测试。该测试的目的是为了验证模型效果没有问题，只是数据差异导致的效果问题。
3. 我对比了本地数据以及KITTI数据，我发现KITTI在Z轴的极值比较低，本地数据Z轴极值较高(地面/车辆效果好，其他效果差)，屏蔽掉较高的点，极值只取到和kitti差不多的高度，是不是可以效果更好(我认为是和特征图的H有关)。
这些问题暂时先不考虑设备安装角度/位置导致的问题，谢谢您的回复！

**Comment by huixiancheng** [comment]
你好！
关于1：该问题确实是球面投影的局限所致，对于同一方位角和顶角，只能保留一个点。关于这个问题，你可以看下[PRNet](https://www.ijcai.org/proceedings/2022/0156.pdf)、[CPGNet](https://arxiv.org/pdf/2204.09914.pdf)和[GFNet](https://arxiv.org/pdf/2207.02605.pdf)都有相关的讨论。理论上，多视角BEV RV PV融合会好点。当然，也可以尝试下多重投影的方法，可以尝试不只保留最前面的点，参考[ThickSeg](https://www.sciencedirect.com/science/article/pii/S0262885621000664)，不过这篇没开源。[RangeFormer](https://arxiv.org/abs/2303.05367)也讨论了这个Many-to-One的问题，理论上Transformer架构也会提升模型的鲁棒性。增大分辨率我不确定能不能解决这种问题， 看上去只能投影完后做插值resize范围图像，这样做应该是会涨点的。(参考[TORNADO-NET](https://arxiv.org/pdf/2008.10544.pdf)（128x2048）和[KPRNet](https://arxiv.org/pdf/2007.12668.pdf)（289x4097）的实验部分)

关于2：是的，我也觉得可以这么做 但是性能应该还是会降的。参考我们Readme之前更新的Robo3D Table 10里Sensor那一排的效果，miou下降的还是比较明显的。不过看上去CPGNet的效果会很好（我猜是BEV下beams数的影响会降低），可以参考下当时复现的[CPGNet](https://github.com/huixiancheng/No-CPGNet)，Robo3D应该是用我之前版本训练的模型测试的。

关于3：这一块的话我不太确定，应为没有接触过其他真实数据，感觉可以尝试下。或者统计下分布Normalized一下hhh 
个人感觉也可以做SSL半监督生成pseudo label 然后fixed权重 finetune `self.conv1 = BasicConv2d(5, 64, kernel_size=3, padding=1)` 这块。当然，如果是类似MINet那样x、y、z、d、r 5通道分开finetune会更容易点。

工程经验有限hhh 可能一些建议没啥参考价值:joy:

**Comment by mp0312** [comment]
太强了
